etcdserver/server.go

```diff
@@ -649,7 +651,7 @@ func (s *EtcdServer) run() {
 		sched.Stop()
 
 		// wait for gouroutines before closing raft so wal stays open
-		s.wg.Wait()
+		s.goWait()
```

Comment:
This breaks `etcdserver/server_test.go TestTriggerSnap` tests since it's returning before `goAttach` triggers snapshot. Then should we change the test? I will investigate more. Thanks.